cmd/nodecmd/setup.go

```diff
@@ -0,0 +1,257 @@
+		if wgResults.HasNodeIDWithError(node.NodeID) {
+			ux.Logger.RedXToUser("Node %s has ERROR: %s", node.IP, wgResults.GetErrorHostMap()[node.IP])
+		}
+	}
+
+	if wgResults.HasErrors() {
+		return fmt.Errorf("failed to deploy node(s) %s", wgResults.GetErrorHostMap())
+	} else {
+		ux.Logger.PrintToUser(logging.Green.Wrap("AvalancheGo and Avalanche-CLI installed and node(s) are bootstrapping!"))
+	}
+	return nil
+}
+
+func setupNode(_ *cobra.Command, _ []string) error {
+	network, err := networkoptions.GetNetworkFromCmdLineFlags(
+		app,
+		"",
+		globalNetworkFlags,
+		false,
+		true,
+		createSupportedNetworkOptions,
+		"",
+	)
+	if err != nil {
+		return err
+	}
+	if network.Kind == models.EtnaDevnet {
+		publicHTTPPortAccess = true // public http port access for etna devnet api for PoAManagerDeployment
+		bootstrapIDs = constants.EtnaDevnetBootstrapNodeIDs
+		bootstrapIPs = constants.EtnaDevnetBootstrapIPs
+
+		// create genesis and upgrade files
+		genesisTmpFile, err := os.CreateTemp("", "genesis")
+		if err != nil {
+			return err
+		}
+		if _, err := genesisTmpFile.Write(constants.EtnaDevnetGenesisData); err != nil {
+			return err
+		}
+		if err := genesisTmpFile.Close(); err != nil {
+			return err
+		}
+		genesisPath = genesisTmpFile.Name()
+
+		upgradeTmpFile, err := os.CreateTemp("", "upgrade")
+		if err != nil {
+			return err
+		}
+		if _, err := upgradeTmpFile.Write(constants.EtnaDevnetUpgradeData); err != nil {
+			return err
+		}
+		if err := upgradeTmpFile.Close(); err != nil {
+			return err
+		}
+		upgradePath = upgradeTmpFile.Name()
+
+		defer func() {
+			_ = os.Remove(genesisTmpFile.Name())
+			_ = os.Remove(upgradeTmpFile.Name())
+		}()
+	}
+	avaGoVersionSetting := node.AvalancheGoVersionSettings{
+		UseAvalanchegoVersionFromSubnet:       useAvalanchegoVersionFromSubnet,
+		UseLatestAvalanchegoReleaseVersion:    useLatestAvalanchegoReleaseVersion,
+		UseLatestAvalanchegoPreReleaseVersion: useLatestAvalanchegoPreReleaseVersion,
+		UseCustomAvalanchegoVersion:           useCustomAvalanchegoVersion,
+	}
+	avalancheGoVersion, err := node.GetAvalancheGoVersion(app, avaGoVersionSetting)
+	if err != nil {
+		return err
+	}
+
+	if !useSSHAgent {
+		if len(nodeIPs) != len(sshKeyPaths) {
+			return fmt.Errorf("--node-ips and --ssh-key-paths should have same number of values")
+		}
+	}
+
+	if err = promptSetupNodes(); err != nil {
+		return err
+	}
+
+	hosts := []*models.Host{}
+	for i, nodeIP := range nodeIPs {
+		sshKeyPath := ""
+		if !useSSHAgent {
+			sshKeyPath = sshKeyPaths[i]
+		}
+		hosts = append(hosts, &models.Host{
+			SSHUser:           constants.RemoteSSHUser,
+			IP:                nodeIP,
+			SSHPrivateKeyPath: sshKeyPath,
+		})
+	}
+	if err = setup(hosts, avalancheGoVersion, network); err != nil {
+		return err
+	}
+	printSetupResults(hosts)
+	return nil
+}
+
+func printSetupResults(hosts []*models.Host) {
+	for _, host := range hosts {
+		nodePath := filepath.Join(app.GetNodesDir(), "staking", host.IP)
```

Comment:
should be abstracted so we can hide if we access staking data from IP or NodeID(for backwards compatibility